### ISSUE TITLE
Default parse behavior swallows 4xx validation errors

### ISSUE THREAD

**Issue by rymohr** [boo1ean/apiapi#9]
The current `parse` behavior swallows any responses outside the 2xx range, but most json api's will return json error responses for the 4xx range.  

What do you think about using something like this for the default instead?

``` js
parse: function(response, body) {
  if (response.statusCode >= 200 && response.statusCode < 300) {
    return body;
  } else if (response.statusCode >= 400 && response.statusCode < 500) {
    throw body;
  }

  throw new Error("Server response status: " + response.statusCode);
}
```

I'm not sure about 3xx. Most of the time they seem to be handled behind the scenes by the request library.

**Comment by boo1ean** [comment]
Need to check redirects, if it works then it's fine.

**Comment by boo1ean** [comment]
Thanks!
I've came up with this version https://github.com/boo1ean/apiapi/blob/master/index.js#L135-L141

Updated and published as 0.4.0

**Comment by rymohr** [comment]
Nice, I hadn't seen lodash's `inRange` helper before.

I'm pretty sure you want to be throwing the body for 4xx responses though.  They're still errors so you expect the promise to be rejected with whatever error response was returned by the server.
